### PR TITLE
Added docker for development support

### PR DESCRIPTION
**Pull request by Mokto**
Allows to have a quick development environment.

`docker-compose -f docker-compose.dev.yml up --build`

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -41,6 +41,5 @@ storage/framework/cache/*
 storage/framework/sessions/*
 storage/framework/views/*
 storage/logs/*
-tests/
```
[the remark this answers is not in the document]
Comment (Mokto) [reply .dockerignore]: Got it ! But it is needed for composer vendors installation.

The main issue in my opinion is this line on the prod Dockerfile.
ADD . .

We should specify which files will be added to the project in the dockerfile, not in the dockerignore.

[the remark this answers is not in the document]
Comment (Mokto) [reply .dockerignore]: I just added phpmyadmin + volumes + removed tests directory from prod dockerfile

[the remark this answers is not in the document]
Comment (asbiin) [reply .dockerignore]: I agree with this. I've listed a specific files/dir list to make the tar.gz in here: https://github.com/monicahq/monica/blob/master/Makefile#L66

File: Dockerfile [hunk 2]
```diff
@@ -46,6 +46,7 @@ WORKDIR /var/www/monica
 # copy system files into their proper homes, and set file ownership
 # correctly
 ADD . .
+RUN rm -rf tests
```
Comment (asbiin) [inline Dockerfile]: Could you move this line in the next `RUN` block? to limit the instruction number.

File: Dockerfile.dev [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ADD .env.dev .env
+RUN chown -R monica:monica . && \
+    chgrp -R apache bootstrap/cache storage && \
+    chmod -R g+w bootstrap/cache storage && \
+    # Apache2 conf
+    cp scripts/docker/000-default.conf /etc/apache2/conf.d/; \
+    # Composer installation
+    scripts/docker/install-composer.sh; \
+    # Set crontab for schedules
+    echo '* * * * * /usr/bin/php /var/www/monica/artisan schedule:run' | crontab -u monica -; \
```
Comment (asbiin) [inline Dockerfile.dev]: I think crontab is not needed for the env docker image (but it's not a problem)

**Comment by djaiss** [comment]
@Mokto have you seen @asbiin 's comments?

File: Dockerfile.dev (hunk 3 above)
Comment (Mokto) [reply Dockerfile.dev]: Won't we ever use this ?

File: Dockerfile (hunk 2 above)
Comment (Mokto) [reply Dockerfile]: I'm not sure to understand why. 

Don't we use https://github.com/jwilder/docker-squash ?

File: .dockerignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mokto) [reply .dockerignore]: What do you prefer ? I'm not really confident modifying the prod docker image TBH.

**Comment by Mokto** [comment]
@djaiss done answering.

File: Dockerfile (hunk 2 above)
Comment (Mokto) [reply Dockerfile]: Or another similar tool

Comment (asbiin) [reply Dockerfile]: @Mokto nope! I was not aware of this tool. Anyway, for this dev image, it doesn't really matter

File: Dockerfile.dev (hunk 3 above)
Comment (asbiin) [reply Dockerfile.dev]: @Mokto I think in dev, the easiest way is to call `php artisan monica:schedule` with command line, by demand.

Comment (theomathieubhvr) [reply Dockerfile.dev]: Ok I'll remove it

Comment (Mokto) [reply Dockerfile.dev]: Ok I'll remove the crons

**Comment by Mokto** [comment]
Done

**Pull request merged**